test/evergreen.yml

```diff
@@ -347,15 +348,57 @@ functions:
         ${test_env_vars|} $CTEST -L check ${smp_command|} --output-on-failure ${check_args|} 2>&1
 
   "cppsuite test":
-    command: shell.exec
-    params:
-      # The tests need to be executed in the cppsuite directory as some required libraries have
-      # their paths defined relative to this directory.
-      working_dir: "wiredtiger/cmake_build/test/cppsuite"
-      script: |
-        set -o errexit
-        set -o verbose
-        ${test_env_vars|} ./run -t ${test_name} -C '${test_config}' -f ${test_config_filename} -l 2
+    - command: shell.exec
+      params:
+        # The tests need to be executed in the cppsuite directory as some required libraries have
+        # their paths defined relative to this directory.
+        # The below script saves the exit code from the test to use it later in this function. By
+        # doing this we can define our own custom artifact upload task without it being cancelled by
+        # the test failing.
+        # Additionally if the test fails perf statistics won't be uploaded as they may be invalid
+        # due to the test failure.
+        working_dir: "wiredtiger/cmake_build/test/cppsuite"
+        shell: bash
+        script: |
+          set -o verbose
+          ${test_env_vars|} ./run -t ${test_name} -C '${test_config}' -f ${test_config_filename} -l 2
+          exit_code=$?
+          echo "$exit_code" > cppsuite_exit_code
+          if [ "$exit_code" != 0 ]; then
+            echo "[{\"info\":{\"test_name\": \"${test_name}\"},\"metrics\": []}]" > ${test_name}.json
+          fi
+          exit 0
+    - command: perf.send
+      params:
+        file: ./wiredtiger/cmake_build/test/cppsuite/${test_name}.json
+    - command: archive.targz_pack
+      params:
+        target: archive.tgz
+        source_dir: wiredtiger/cmake_build/test/cppsuite
+        include:
+          - "./**"
+    - command: s3.put
+      params:
+        aws_secret: ${aws_secret}
+        aws_key: ${aws_key}
+        local_file: archive.tgz
+        bucket: build_external
+        permissions: public-read
+        content_type: application/tar
+        display_name: cppsuite-test
+        remote_file: wiredtiger/${build_variant}/${revision}/artifacts/${task_name}_${build_id}${postfix|}.tgz
+    - command: shell.exec
+      params:
+        shell: bash
+        script: |
+          set -o verbose
+          if [ -f wiredtiger/cmake_build/test/cppsuite/cppsuite_exit_code ]; then
+            exit_code=`cat wiredtiger/cmake_build/test/cppsuite/cppsuite_exit_code`
+          else
+            exit_code=0
+          fi
+          rm -rf wiredtiger
```

Comment:
Is it worth adding a comment to describe the reason for the folder removal?

Reply:
Will do.

Reply:
78893f624